File: 3Ain/TPS/impiccato/backup/impiccato.pas
program impiccato;

{$mode objfpc}{$H+}

uses
  {$IFDEF UNIX}{$IFDEF UseCThreads}
  cthreads,
  {$ENDIF}{$ENDIF}
  Classes
  { you can add units after this };
type
    str=array[0..100] of char;
var
    i,sum,x:integer;
    let:char;
    piumeno,parola:str;
procedure translate(parola,piumeno:str);
        begin
	for i:=0 to length(parola) do
                begin
		if (ord(parola[i])=97) or (ord(parola[i])=101) or (ord(parola[i])=105) or (ord(parola[i])=111) or (ord(parola[i])=117) then
			ord(piumeno[i]):=43;
                if (ord(parola[i])<>97) or (ord(parola[i])<>101) or (ord(parola[i])<>105) or (ord(parola[i])<>111) or (ord(parola[i])<>117) then
			ord(piumeno[i]):=45;
	        end;
        end;
procedure cmp(parola,piumeno:str; let:char);
        begin
	for i:=0 to length(parola) do
		if parola[i]=let then
			piumeno[i]:=let;
        end;
procedure checkwin(parola,piumeno:str; x,sum:integer);
        begin
        sum:=0;
	for i:=0 to length(piumeno) do
		if parola[i]=piumeno[i] then
			sum+=1;
	if sum=length(parola) then
		x:=0;
	if sum<>length(parola) then
		x:=1;
        end;
begin
	parola:='segreto';
	translate(parola,piumeno);
	writeln(piumeno);
        sum:=0;
	x:=1;
	while(x=1) do
        begin
	readln(let);
	cmp(parola,piumeno,let);
	writeln(piumeno);
	checkwin(parola,piumeno,x,sum);
	end;
        readln();
end.

File: 3Ain/Informatica/impiccato.c
#include<stdio.h>
#include<string.h>
#include<ctype.h>
#include<conio.h>
void translate(char parola[50],char piumeno[50]){
	for(size_t i=0;i<strlen(parola);i++){
		if(parola[i]==97 || parola[i]==101 || parola[i]==105 || parola[i]==111 || parola[i]==117){
			piumeno[i]=43;
		}
		else{
			piumeno[i]=45;
		}
	}
}
void cmp(char parola[50],char piumeno[50],char let){
	for(size_t i=0;i<strlen(parola);i++){
		if(parola[i]==let){
			piumeno[i]=let;
		}
	}
}
int checkwin(char parola[50],char piumeno[50],int x){
	int sum=0;
	for(size_t i=0;i<strlen(piumeno);i++){
		if(parola[i]==piumeno[i]){
			sum+=1;
		}
	}
	if(sum==strlen(parola)){
		return 0;
	}
	else{
		return 1;	
	}
}
int main(){
	char parola[]="segreto";
	char piumeno[strlen(parola)];
	translate(parola,piumeno);
	printf("%s\n",piumeno);
	int j,x=1;
	while(x==1){
	char let;
	scanf("%c",&let);
	cmp(parola,piumeno,let);
	printf("%s\n",piumeno);
	checkwin(parola,piumeno,x);
	}
}
